### PR TITLE
Clarify the vector files utilized for preloading

### LINKED ISSUE #116273: Clarify vector preloading docs
In our optimize for kNN docs page, we talk about pre-loading vector files. There and any other place we talk about vector preloading, we need to clarify which vector files are for what and how to preload only files that correspond to quantized vectors if quantization is enabled.

### PR DESCRIPTION
**Pull request by benwtrent**
Adds clarification for vector preloading, what extension is to what storage kind, and that quantized vectors are stored in separate files allowing for individual preload. 

closes: https://github.com/elastic/elasticsearch/issues/116273

### CODE REVIEW

File: docs/reference/how-to/knn-search.asciidoc [hunk 1]
```diff
@@ -97,13 +97,19 @@ and https://elasticsearch-benchmarks.elastic.co/#tracks/dense_vector[here] some
...
 The following file extensions are used for the approximate kNN search:
+Each extension is broken down by the quantization types.
 
-* `vec` and `veq` for vector values
-* `vex` for HNSW graph
-* `vem`, `vemf`, and `vemq` for metadata
+* `vec` for all non-quantized vector values. This includes all element types: `float`, `byte`, and `bit`.
+* `veq` for quantized vectors indexed with <<dense-vector-quantization,`quantization`>>: `int4` or `int8`
+* `veb` for binary vectors indexed with <<dense-vector-quantization,`quantization`>>: `bbq`
+* `vem`, `vemf`, `vemq`, and `vemb` for metadata, usually small and not a concern for preloading
```
Comment (pmpailis) [inline docs/reference/how-to/knn-search.asciidoc:109]: Is there a reason for leaving `vex` out of this list, or was it just overlooked?

**Review by pmpailis — APPROVED** [review]
LGTM ; only a minor comment on a field type

File: docs/reference/how-to/knn-search.asciidoc (hunk 1 above)
Comment (benwtrent) [reply docs/reference/how-to/knn-search.asciidoc:109]: Ah, it was indeed overlooked, let me add that.

File: docs/reference/how-to/knn-search.asciidoc [hunk 2]
```diff
@@ -97,13 +97,19 @@ and https://elasticsearch-benchmarks.elastic.co/#tracks/dense_vector[here] some
...
+[[dense-vector-preloading]]
 include::search-speed.asciidoc[tag=warm-fs-cache]
 
 The following file extensions are used for the approximate kNN search:
+Each extension is broken down by the quantization types.
 
-* `vec` and `veq` for vector values
-* `vex` for HNSW graph
-* `vem`, `vemf`, and `vemq` for metadata
+* `vec` for all non-quantized vector values. This includes all element types: `float`, `byte`, and `bit`.
```
Comment (benwtrent) [inline docs/reference/how-to/knn-search.asciidoc:106]:
```suggestion
* `vex` for the HNSW graph
* `vec` for all non-quantized vector values. This includes all element types: `float`, `byte`, and `bit`.
```

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**